components/automate-ui/src/app/pages/project/details/project-details.component.spec.ts

```diff
@@ -0,0 +1,199 @@
+        MockComponent({ selector: 'chef-loading-spinner' }),
+        MockComponent({ selector: 'chef-option' }),
+        MockComponent({ selector: 'chef-page-header' }),
+        MockComponent({ selector: 'chef-subheading' }),
+        MockComponent({ selector: 'chef-table' }),
+        MockComponent({ selector: 'chef-thead' }),
+        MockComponent({ selector: 'chef-tbody' }),
+        MockComponent({ selector: 'chef-tr' }),
+        MockComponent({ selector: 'chef-th' }),
+        MockComponent({ selector: 'chef-td' }),
+        MockComponent({ selector: 'ng-container', inputs: ['hidden'] }),
+        ProjectDetailsComponent
+      ],
+      imports: [
+        ReactiveFormsModule,
+        RouterTestingModule,
+        ChefPipesModule,
+        StoreModule.forRoot({
+          router: () => ({
+            state: {
+              url: '/projects/uuid-1',
+              queryParams: {},
+              params: {},
+              fragment: '',
+              path: ['/']
+            },
+            previousRoute: {},
+            navigationId: 0
+          }),
+          projects: projectEntityReducer
+        })
+      ],
+      providers: [
+        FeatureFlagsService
+      ]
+    }).compileComponents();
+  }));
+
+  let store: Store<NgrxStateAtom>;
+  beforeEach(() => {
+    store = TestBed.get(Store);
+
+    store.dispatch(new GetProjectSuccess({
+      project: {
+        id: 'uuid-1', name: 'Default',
+        type: <IAMType>'CHEF_MANAGED'
+      }
+    }));
+
+    jasmine.addMatchers(customMatchers);
+    fixture = TestBed.createComponent(ProjectDetailsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  describe('when there are no rules', () => {
+    beforeEach(() => {
+      store.dispatch(new GetRulesSuccess({
+          rules: []
+      }));
+      component.rules = [];
+      fixture.detectChanges();
+    });
+
+    it('rules array should be empty', () => {
+      expect(component.rules.length).toBe(0);
+    });
+
+    it('show rules section when rules tab is selected', () => {
+      component.onTabChange({ target: { value: 'rules'} });
+      expect(component.showTab('rules')).toBeTruthy();
+    });
+
+    it('should not display rule table when no rules', () => {
+      expect(component.showRulesTable()).toBeFalsy();
+    });
+
+    it('the create your first rule message should display', () => {
+      expect(component.showFirstRuleMessage()).toBeTruthy();
+    });
+  });
+
+  describe('when there are rules', () => {
+    beforeEach(() => {
+      store.dispatch(new GetRulesSuccess({
+          rules: rules
+      }));
+      component.project = project;
+      component.rules = rules;
+      fixture.detectChanges();
+    });
+
+    it('rules array should have two rules', () => {
+      expect(component.rules.length).toBe(2);
+    });
+
+    it('show rules section when rules tab is selected', () => {
+      component.onTabChange({ target: { value: 'rules'} });
```

Comment:
This falls under the category of what I call "phantom tests". If you delete L179 the test still passes. So a "pass" is telling us nothing; here's a suggestion on how to make it tell us "something".
```
it('shows/hides rules section when based on selection', () => {
      component.onTabChange({ target: { value: 'details'} });
      expect(component.showTab('rules')).toBeFalsy();
      component.onTabChange({ target: { value: 'rules'} });
      expect(component.showTab('rules')).toBeTruthy();
    });
```
And I would do the analogous test for the details section.

(These comments applies to the prior section on "no rules", too.)

(Reference on phantom tests: [The Phantom Menace in Unit Testing](https://www.red-gate.com/simple-talk/dotnet/software-testing/the-phantom-menace-in-unit-testing/))